src/bigquery/base.py

```diff
@@ -0,0 +1,108 @@
+#
+"""Implementation Bigquery builder base."""
+import os
+import json
+import time
+import tempfile
+from rudra import logger
+from src.config.settings import GCP_SETTINGS
+from google.cloud.bigquery.job import QueryJobConfig
+from google.cloud.bigquery.client import Client
+
+
+class BigqueryBuilder:
+    """BigqueryBuilder class Implementation."""
+
+    def __init__(self):
+        """Initialize the BigqueryBuilder object."""
+        logger.info('Storing BigQuery Auth Credentials')
+        key_file_contents = {
+            "type": GCP_SETTINGS.type,
+            "project_id": GCP_SETTINGS.project_id,
+            "private_key_id": GCP_SETTINGS.private_key_id,
+            "private_key": GCP_SETTINGS.private_key,
+            "client_email": GCP_SETTINGS.client_email,
+            "client_id": GCP_SETTINGS.client_id,
+            "auth_uri": GCP_SETTINGS.auth_uri,
+            "token_uri": GCP_SETTINGS.token_uri,
+            "auth_provider_x509_cert_url": GCP_SETTINGS.auth_provider_x509_cert_url,
+            "client_x509_cert_url": GCP_SETTINGS.client_x509_cert_url,
+        }
+        tfile = tempfile.NamedTemporaryFile(mode='w+', delete=True)
+        tfile.write(json.dumps(key_file_contents))
+        tfile.flush()
+        tfile.seek(0)
+        self.credential_path = tfile.name
+
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = self.credential_path
+        logger.info('GOOGLE_APPLICATION_CREDENTIALS %s', self.credential_path)
+
+        logger.info('Creating new query job configuration')
+        self.query_job_config = QueryJobConfig()
+        self.client = Client(default_query_job_config=self.query_job_config)
+        self.job_query_obj = None
+
+        tfile.close()
+
+    def run_query_sync(self):
+        """Run the bigquery synchronously."""
+        if self.client and self.query:
+            self.job_query_obj = self.client.query(
+                self.query, job_config=self.query_job_config)
+            while not self.job_query_obj.done():
+                time.sleep(0.1)
+            return self.job_query_obj.job_id
+        else:
+            raise ValueError('Client or query missing')
+
+    def get_result(self):
+        """Get the result of the job."""
+        if self.job_query_obj:
+            for row in self.job_query_obj.result():
+                yield ({k: v for k, v in row.items()})
+        else:
+            raise ValueError('Job is not initialized')
+
+
+class DataProcessing:
+    """Process the Bigquery Data."""
+
+    def __init__(self, s3_client=None):
+        """Initialize DataProcessing object."""
+        self.s3_client = s3_client
+
+    def update_s3_bucket(self, data,
+                         bucket_name,
+                         filename='collated.json'):
+        """Upload s3 bucket."""
+        # connect after creating or with existing s3 client
+        self.s3_client.connect()
+        if not self.s3_client.is_connected():
+            raise ValueError('Unable to connect to s3.')
```

Comment:
Why is ValueError being raised for all exceptions?

Reply:
This was case with all AWS exceptions in Rudra, and I followed the same. @prashbnair Let me know if you have any better alternative.